### PR TITLE
dev: add send_and_resolve

### LINKED ISSUE #10845: Add PayloadStore::send_and_resolve helper function
### Describe the feature

currently we have two functions for

https://github.com/paradigmxyz/reth/blob/f6444b2e704aab725320580104e1f97e68bdf9e1/crates/payload/builder/src/service.rs#L154-L165

and

https://github.com/paradigmxyz/reth/blob/f6444b2e704aab725320580104e1f97e68bdf9e1/crates/payload/builder/src/service.rs#L117-L128

it would be great if we have a function that does this in one go

https://github.com/paradigmxyz/reth/blob/f6444b2e704aab725320580104e1f97e68bdf9e1/crates/payload/builder/src/traits.rs#L22-L26

send attributes and resolve right away, returning a future that resolves to the builtpayload

cc @greged93 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by greged93**
Adds a send and resolve helper on the `PayloadBuilderHandle`. This helper allows to request for a payload to be build based on the provided payload builder attribute and returns a future that resolves to the payload.

Resolves #10845

### CODE REVIEW

File: crates/payload/builder/src/service.rs [hunk 1]
```diff
@@ -28,6 +28,8 @@ use tokio_stream::wrappers::UnboundedReceiverStream;
 use tracing::{debug, info, trace, warn};
 
 type PayloadFuture<P> = Pin<Box<dyn Future<Output = Result<P, PayloadBuilderError>> + Send + Sync>>;
+type MaybePayloadFuture<P> =
+    Pin<Box<dyn Future<Output = Option<Result<P, PayloadBuilderError>>> + Send + Sync>>;
```
Comment (mattsse) [inline crates/payload/builder/src/service.rs:32]: I don't think we need this

File: crates/payload/builder/src/service.rs [hunk 2]
```diff
@@ -127,6 +129,31 @@ where
         }
     }
 
+    /// Sends a message to the service to start building a new payload for the given payload
+    /// attributes and returns a future that resolves to the payload.
+    pub fn send_and_resolve_payload(
```
Comment (mattsse) [inline crates/payload/builder/src/service.rs:134]: this can just be an async function

File: crates/payload/builder/src/service.rs [hunk 3]
```diff
@@ -127,6 +129,31 @@ where
...
+                Ok(Err(e)) => return Some(Err(e)),
+                Ok(Ok(id)) => id,
+            };
+
+            let (tx, rx) = oneshot::channel();
+            to_service.send(PayloadServiceCommand::Resolve(id, tx)).ok()?;
+            match rx.await.transpose()? {
+                Ok(fut) => Some(fut.await),
+                Err(e) => Some(Err(e.into())),
+            }
```
Comment (mattsse) [inline crates/payload/builder/src/service.rs:153]: then you can do all of this without the Box::pin

**Review by mattsse — CHANGES_REQUESTED** [review]
one suggestion

File: crates/payload/builder/src/service.rs (hunk 3 above)
Comment (greged93) [reply crates/payload/builder/src/service.rs:153]: Ah ok I thought you wanted to return a future, will update

Comment (mattsse) [reply crates/payload/builder/src/service.rs:153]: yeah but you can return Option<PayloadFuture>

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**